frontend/test/__support__/server-mocks/collection.ts

```diff
@@ -72,3 +72,97 @@ export function setupUnauthorizedCollectionsEndpoints(
 ) {
   collections.forEach(setupUnauthorizedCollectionEndpoints);
 }
+
+export function setupCollectionByIdEndpoint({
+  collections,
+  error,
+}: {
+  collections: Collection[];
+  error?: string;
+}) {
+  if (error) {
+    setupCollectionWithErrorById({ error });
+    return;
+  }
+
+  fetchMock.get(/api\/collection\/\d+/, url => {
+    const collectionIdParam = url.split("/")[5];
+    const collectionId = Number(collectionIdParam);
+
+    const collection = collections.find(
+      collection => collection.id === collectionId,
+    );
+
+    return collection;
+  });
+}
+
+function setupCollectionWithErrorById({
+  error,
+  status = 500,
+}: {
+  error: string;
+  status?: number;
+}) {
+  fetchMock.get(/api\/collection\/\d+/, () => {
+    return {
+      body: error,
+      status,
+    };
+  });
+}
+
+export function setupRootCollectionItemsEndpoint({
+  collections = [],
+  dashboards = [],
+}: {
+  collections: Collection[];
+  dashboards: Dashboard[];
+}) {
+  fetchMock.get("path:/api/collection/root/items", () => {
+    const rootDashboards = dashboards.filter(
+      dashboard => dashboard.collection_id === null,
+    );
+    const rootCollections = collections.filter(
+      collection => collection.location !== "/",
+    );
+    const data = [...rootDashboards, ...rootCollections];
+
+    return {
+      total: data.length,
+      data,
+    };
+  });
+}
+
+export function setupCollectionItemsEndpoint({
+  dashboards = [],
+}: {
+  dashboards: Dashboard[];
+}) {
+  fetchMock.get(/api\/collection\/\d+\/items/, url => {
+    const collectionIdParam = url.split("/")[5];
+
+    const collectionId = Number(collectionIdParam);
+
+    const dashboardsOfCollection = dashboards.filter(
+      dashboard => dashboard.collection_id === collectionId,
+    );
+
+    return {
+      total: dashboardsOfCollection.length,
+      data: dashboardsOfCollection,
+    };
+  });
+}
+
+export function setupPersonalCollectionItemsEndpont({
```

Comment:
Also the same as the previous one - it's ok to have `Collection` with `id = "personal" and in fact personal collections are implemented as regular collections https://github.com/metabase/metabase/blob/e0d4723f09b7695db55596c2d794d015cb124b7f/frontend/src/metabase-types/api/collection.ts#L7, so there should be just one helper